Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs

```diff
@@ -659,9 +659,12 @@ public static List<TreeViewItemData<ActionOrBindingData>> GetActionsAsTreeViewDa
 
         private static int GetIdForGuid(Guid guid, Dictionary<Guid, int> idDictionary)
         {
+            //This method is used to ensure that the same Guid always gets the same id
+            //We use getHashCode instead of a counter, as we cannot guarantee that the same Guid will always be added in the same order
+            //There is a tiny chance of a collision, but it is it does happen it will only affect the expanded state of the tree view
             if (!idDictionary.TryGetValue(guid, out var id))
             {
-                id = idDictionary.Values.Count > 0 ? idDictionary.Values.Max() + 1 : 0;
+                id = guid.GetHashCode(); //idDictionary.Values.Count > 0 ? idDictionary.Values.Max() + 1 : 0;
```

Comment:
Thanks for pointing out the chance of collision here. Maybe I am overlooking something but why don't we just use Guid as ID instead of converting to an int (with risk of collision?

Would recommend inserting space above after the // in the inline comments and also to remove the commented out code.

Reply:
The reason for the guid -> int id mapping is that that the UIElements TreeViewItem requires an int id.